<p>Conditional blocks of code should do an early return whenever possible. Early returns improve
the code's performance and legibility</p>

<h2>Noncompliant Code Example</h2>
<pre>
  public boolean isPositive(int number) {
      if (number > 0) { // Noncompliant
        return true;
      } else {
        return false;
      }
  }
</pre>

<h2>Compliant Solution</h2>
<pre>
  public boolean isPositive(int number) {
      if (number <= 0) // Compliant
        return false;
      return true;
  }
</pre>