<p>When making queries using OBCriteria, if only the first result is going to be used, then there's no need for
  the <i>OBCriteria.list()</i> method to be used only to then get the first entry on the resulting list.
  The methods <i>OBCriteria.setMaxResults(1)</i> and <i>OBCriteria.uniqueResult()</i> should be used instead.</p>

<h2>Noncompliant Code Example</h2>
<pre>
  public Product getInvoiceUsingList(String invoiceId) {
    OBCriteria&ltInvoice&gt invCriteria = OBDal.getInstance().createCriteria(Invoice.class);
    invCriteria.add(Restrictions.eq(Invoice.PROPERTY_ID, invoiceId));

    return invCriteria.list().get(0); // Noncompliant
  }
</pre>

<h2>Compliant Solution</h2>
<pre>
  public Product getProductUsingUnique(String prodId) {
    OBCriteria&ltProduct&gt prodCriteria = OBDal.getInstance().createCriteria(Product.class);
    prodCriteria.add(Restrictions.eq(Product.PROPERTY_ID, prodId));
    prodCriteria.setMaxResults(1);

    return prodCriteria.uniqueResult(); // Compliant
  }
</pre>