<p>Loop termination conditions should never use the <i>OBCriteria.list()</i> method inside the clause directly,
  because this could greatly affect execution times.
  Instead, a variable should be initialized, and the method invocation should be put as its value.</p>

<h2>Noncompliant Code Example</h2>
<pre>
  public void processOrderHist() {
      OBCriteria&ltOrderHistoryItem&gt criteria = OBDal.getInstance().createCriteria(OrderHistoryItem.class);
      for (OrderHistory item : criteria.list()) { // Noncompliant
        // ...
      }
  }
</pre>

<h2>Compliant Solution</h2>
<pre>
  public void processOrderHist() {
      OBCriteria&ltOrderHistoryItem&gt criteria = OBDal.getInstance().createCriteria(OrderHistoryItem.class);
      List&ltOrderHistoryItem&gt histItemList = criteria.list();
      for (OrderHistory item : histItemList) { // Compliant
        // ...
      }
  }
</pre>